openmetadata-service/src/main/java/org/openmetadata/csv/EntityCsv.java

```diff
@@ -744,6 +749,20 @@ protected void createEntity(CSVPrinter resultsPrinter, CSVRecord csvRecord, T en
     }
   }
 
+  private void createChangeEventAndUpdateInES(PutResponse<T> response, String importedBy) {
+    if (!response.getChangeType().equals(EventType.ENTITY_NO_CHANGE)) {
+      ChangeEvent changeEvent =
+          FormatterUtil.createChangeEventForEntity(
+              importedBy, response.getChangeType(), response.getEntity());
+      Object entity = changeEvent.getEntity();
+      changeEvent = copyChangeEvent(changeEvent);
+      changeEvent.setEntity(JsonUtils.pojoToMaskedJson(entity));
+      // Change Event and Update in Es
+      Entity.getCollectionDAO().changeEventDAO().insert(JsonUtils.pojoToJson(changeEvent));
```

Comment:
This can block quite a bit. If changeEvent table is busy we will blocking here , especially if there are lot of alerts reading from it.
can we make this async and do it for all the responses at one go , instead of calling it one at a time.

Reply:
made it async